Fix linkage issues on X11 (https://github.com/NixOS/nixpkgs/issues/142583).

GLFW dynamically loads X11 libraries. On Nix they are not on the default
search path, so link them so they appear in DT_NEEDED.

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -211,6 +211,16 @@
         message(FATAL_ERROR "X Shape headers not found; install libxext development package")
     endif()
     target_include_directories(glfw PRIVATE "${X11_Xshape_INCLUDE_PATH}")
+
+    target_link_libraries(glfw PRIVATE ${X11_Xrandr_LIB}
+                                       ${X11_Xinerama_LIB}
+                                       ${X11_X11_xcb_LIB}
+                                       ${X11_Xrender_LIB}
+                                       ${X11_Xkb_LIB}
+                                       ${X11_Xcursor_LIB}
+                                       ${X11_Xi_LIB}
+                                       ${X11_Xxf86vm_LIB}
+                                       ${X11_Xext_LIB})
 endif()
 
 if (UNIX AND NOT APPLE)
